### PR TITLE
fix: add better seeded data

### PR DESCRIPTION
**Pull request by ludtkemorgan**
# Pull Request Template

## Issue Overview

This PR addresses #3466 

- [ ] This change addresses the issue in full
- [x] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

As part of the typeORM -> Prisma changeover we need to re-think how data is seeded for both manual and automated testing. This PR creates a new pattern compared to the current backend by doing the following:

- Setup the seed file to do different things based on environment:
  - Prod: Only set the minimum fields needed to have a running environment such as a jurisdiction
  - Stage: Set up data that would make a demo (staging) site usable by a user. It sets up all of the mandatory fields along with mock tables that are consistent every time it is run. Lots of this data is setup from objects with a name and a diverse set  of values.
  - Dev: Sets up values to be used in local testing with completely random values. Names are random and a mix of values are randomly chosen.
- For all environments a jurisdiction name can be passed in the command by adding `--jurisdictionName <name>` when running the command.
  - dev environment is setup with the jurisdiction name of "Bloomington" so that running it locally you can have the name set in the .env file
- For each table type there is a factory to generate rows into that table. Everything that can be randomized is along with the ability to override values.
- The factories are used for all the seed setups (with different override values sent in all cases) and within the tests. The randomized values are then used to validate the fields rather than relying on a set value

## How Can This Be Tested/Reviewed?

After `yarn install` in the `backend_new` directory you can run the following commands and look to see what is in the DB.
`yarn db:setup` - setups the database with completely random values
`yarn db:setup:staging` - setups the database with the pre-defined staging values

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [ ] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: backend_new/prisma/seed-helpers/listing-factory.ts [hunk 1]
```diff
@@ -348,7 +492,8 @@ export const listingFactory = (
   ),
 });
 
-const unitFactory = (
+// Temporarily keeping this for reference
+const unitFactory2 = (
```
Comment (YazeedLoonat) [inline backend_new/prisma/seed-helpers/listing-factory.ts:496]: do we still need this since we have a dedicated unit factory?

**Review by YazeedLoonat — CHANGES_REQUESTED** [review]
Looks good, just one question

File: backend_new/prisma/seed-helpers/listing-factory.ts (hunk 1 above)
Comment (ludtkemorgan) [reply backend_new/prisma/seed-helpers/listing-factory.ts:496]: I was keeping this here so merge conflicts were easier every time I pulled in your changes. I can remove it now that it is ready!

Comment (ludtkemorgan) [reply backend_new/prisma/seed-helpers/listing-factory.ts]: @YazeedLoonat This has been deleted

File: backend_new/prisma/seed-helpers/listing-factory.ts [hunk 2]
```diff
@@ -1,415 +1,147 @@
...
-      county: `listingsBuildingAddress: ${i} county: ${i}`,
-      state: `listingsBuildingAddress: ${i} state: ${i}`,
-      street: `listingsBuildingAddress: ${i} street: ${i}`,
-      street2: `listingsBuildingAddress: ${i} street2: ${i}`,
-      zipCode: `listingsBuildingAddress: ${i} zipCode: ${i}`,
-      latitude: i * 100,
-      longitude: i * 101,
-    },
+  prismaClient: PrismaClient,
+  optionalParams?: {
```
Comment (ColinBuyck) [inline backend_new/prisma/seed-helpers/listing-factory.ts:15]: Nice!

**Review by ColinBuyck — APPROVED** [review]
This LGTM 🌸 just two non-blocking thoughts. 1) In the listing data, could we vary the status? So there are active and closed listings as well. 2) Could we vary the address table data so that we can verify that the address, application pickup address, application drop off address, and application mailing address are all correctly mapping to unique values.

**Review by emilyjablonski — APPROVED** [review]
This LGTM just some general feedback / questions!
* The issue indicates`Product to make some test listings.` was that part of this process?
* Something that has always been really useful to me about the dev data is that it is consistent. If I need to test the state of preferences on a listing that doesn't have any, it's been really nice to have that dev data consistent and I know a listing already exists named something like "No Preferences Listing". I'm curious for your perspective on that, I'm a little worried listing that functionality is something that could slow me down in the future.
* It seems like this could lead to a number of missing states - I think it's just hard to nail them all down - maybe part of this is as new states come up not in the data that block dev we can have license to adjust the factories?

**Review by YazeedLoonat — APPROVED** [review]
LGTM

**Pull request merged**